runtime/services/catalog/artifacts/yaml/objects.go

```diff
@@ -23,27 +23,28 @@ import (
-	Timeout               int32          `yaml:"timeout,omitempty"`
-	Format                string         `yaml:"format,omitempty" mapstructure:"format,omitempty"`
-	Extract               map[string]any `yaml:"extract,omitempty" mapstructure:"extract,omitempty"`
-	DuckDBProps           map[string]any `yaml:"duckdb,omitempty" mapstructure:"duckdb,omitempty"`
-	Headers               map[string]any `yaml:"headers,omitempty" mapstructure:"headers,omitempty"`
-	AllowSchemaRelaxation *bool          `yaml:"ingest.allow_schema_relaxation,omitempty" mapstructure:"allow_schema_relaxation,omitempty"`
-	SQL                   string         `yaml:"sql,omitempty" mapstructure:"sql,omitempty"`
-	DB                    string         `yaml:"db,omitempty" mapstructure:"db,omitempty"`
-	ProjectID             string         `yaml:"project_id,omitempty" mapstructure:"project_id,omitempty"`
+	Type                        string
+	Path                        string         `yaml:"path,omitempty"`
+	CsvDelimiter                string         `yaml:"csv.delimiter,omitempty" mapstructure:"csv.delimiter,omitempty"`
+	URI                         string         `yaml:"uri,omitempty"`
+	Region                      string         `yaml:"region,omitempty" mapstructure:"region,omitempty"`
+	S3Endpoint                  string         `yaml:"endpoint,omitempty" mapstructure:"endpoint,omitempty"`
+	GlobMaxTotalSize            int64          `yaml:"glob.max_total_size,omitempty" mapstructure:"glob.max_total_size,omitempty"`
+	GlobMaxObjectsMatched       int            `yaml:"glob.max_objects_matched,omitempty" mapstructure:"glob.max_objects_matched,omitempty"`
+	GlobMaxObjectsListed        int64          `yaml:"glob.max_objects_listed,omitempty" mapstructure:"glob.max_objects_listed,omitempty"`
+	GlobPageSize                int            `yaml:"glob.page_size,omitempty" mapstructure:"glob.page_size,omitempty"`
+	BatchSize                   string         `yaml:"batch_size,omitempty" mapstructure:"batch_size,omitempty"`
+	HivePartition               *bool          `yaml:"hive_partitioning,omitempty" mapstructure:"hive_partitioning,omitempty"`
+	Timeout                     int32          `yaml:"timeout,omitempty"`
+	Format                      string         `yaml:"format,omitempty" mapstructure:"format,omitempty"`
+	Extract                     map[string]any `yaml:"extract,omitempty" mapstructure:"extract,omitempty"`
+	DuckDBProps                 map[string]any `yaml:"duckdb,omitempty" mapstructure:"duckdb,omitempty"`
+	Headers                     map[string]any `yaml:"headers,omitempty" mapstructure:"headers,omitempty"`
+	AllowSchemaRelaxation       *bool          `yaml:"allow_schema_relaxation,omitempty" mapstructure:"allow_schema_relaxation,omitempty"`
+	IngestAllowSchemaRelaxation *bool          `yaml:"ingest.allow_schema_relaxation,omitempty" mapstructure:"allow_schema_relaxation,omitempty"`
```

Comment:
`mapstructure` also needs to be changed to `ingest.allow_schema_relaxation`. Seems like it is incorrect in main branch.